src/routes/auth/index.ts

```diff
@@ -1,140 +1,286 @@
+			}
+			throw error;
+		}
 
-authRoutes.post('/sign-in', async (context: Context<EnvironmentBindings>) => {
-	let parsedBody: SignInData;
+		const { id: profileId, password: storedPassword } = await getProfileIdPassword(context.env.database, parsedBody.email);
+		if (!profileId || !storedPassword) {
+			return context.json({ message: 'Invalid email' } satisfies StatusResponse, StatusCodes.UNAUTHORIZED);
+		}
 
-	try {
-		const body = await context.req.json();
-		parsedBody = SignInSchema.parse(body);
-	} catch (error) {
-		if (error instanceof SyntaxError) {
-			return context.json<StatusResponse>({ message: `Invalid JSON format: ${error.message}` }, StatusCodes.BAD_REQUEST);
+		const isProfileValid = await checkProfile(context.env.database, profileId);
+		if (!isProfileValid) {
+			return context.json({ message: 'Invalid profile id or Account not activated' } satisfies StatusResponse, StatusCodes.UNAUTHORIZED);
 		}
-		throw error;
-	}
 
-	const { id: profileId, password: storedPassword } = await getProfileIdPassword(context.env.database, parsedBody.email);
-	if (!profileId || !storedPassword) {
-		return context.json<StatusResponse>({ message: 'Invalid email' }, StatusCodes.UNAUTHORIZED);
-	}
+		const accessLevel = await getAccessLevel(context.env.database, profileId);
+		if (!accessLevel) {
+			return context.json({ message: 'Access not found' } satisfies StatusResponse, StatusCodes.NOT_FOUND);
+		}
 
-	const isProfileValid = await checkProfile(context.env.database, profileId);
-	if (!isProfileValid) {
-		return context.json<StatusResponse>({ message: 'Invalid profile id or Account not activated' }, StatusCodes.UNAUTHORIZED);
-	}
+		const isValid = await validatePassword(parsedBody.password, storedPassword);
+		if (!isValid) {
+			return context.json({ message: 'Invalid email or password' } satisfies StatusResponse, StatusCodes.UNAUTHORIZED);
+		}
 
-	const accessLevel = await getAccessLevel(context.env.database, profileId);
-	if (!accessLevel) {
-		return context.json<StatusResponse>({ message: 'Access not found' }, StatusCodes.NOT_FOUND);
-	}
+		const sessionToken = crypto.randomUUID();
+		const hoursOffset = 24;
+		const tokenExpiryISO = addHours(new Date(), hoursOffset).toISOString();
+		const sessionDataObject: SessionData = {
+			id: profileId,
+			email: parsedBody.email,
+			access: accessLevel,
+			expiry: tokenExpiryISO
+		};
+		const sessionData = JSON.stringify(sessionDataObject);
+		await context.env.SESSION_TOKENS.put(sessionToken, sessionData);
 
-	const isValid = await validatePassword(parsedBody.password, storedPassword);
-	if (!isValid) {
-		return context.json<StatusResponse>({ message: 'Invalid email or password' }, StatusCodes.UNAUTHORIZED);
+		context.header('Set-Cookie', `auth_token=${sessionToken}; HttpOnly; Secure; SameSite=Strict; Expires=${tokenExpiryISO}; Path=/;`);
+
+		return context.json({ message: 'Sign in successful' } satisfies StatusResponse, StatusCodes.CREATED);
```

Comment:
I don't remember in which PR exactly, but there is a comment to change this specific call to use the hono cookie helper.

Reply:
Yeah we'll take care of this in the logout PR